addons/survey/controllers/main.py

```diff
@@ -407,7 +407,7 @@ def survey_submit(self, survey_token, answer_token, **post):
                 go_back = False
                 answer_sudo._send_certification()
                 vals = {'state': 'done'}
-            else:
+            elif 'button_submit' in post:
```

Comment:
here to minimize risk (if people did not have `button_submit`) it would be better to keep `else:` and have before `elif 'breadcrum_redirect' in post:`

on the `ret['redirect'] = '/survey/fill/%s/%s' % (survey_sudo.access_token, answer_token)` you can just replace with `res.setdefault('redirect', thevalue)`